thorlcr/activities/indexread/thindexreadslave.cpp

```diff
@@ -719,10 +710,66 @@ class CIndexReadSlaveBase : public CSlaveActivity
     virtual void gatherActiveStats(CRuntimeStatisticCollection &activeStats) const
     {
         PARENT::gatherActiveStats(activeStats);
+        // *) jhtree stats will have been tracked as follows
+        //    - For superfiles, there will be 1 IContextLogger in contextLoggers[] for each subfile
```

Comment:
Could there be a very large number of parts?  If that is the case, do you think the performance would be impacted enough to make the change worthwhile?

I'd like leave as it is unless it is really worthwhile. Although change looks trivial, it will require significant time consuming re-testing.

Reply:
could be quite large, e.g. a superfile of 50+ files of 400x files, but each workers only gets a slice of those, and these CStatsContextLogger objects aren't that big.
I would merge as is, and perhaps open a low-prio. JIRA to revisit it on a rainy day.

Reply:
I've created a jira to revisit this: https://track.hpccsystems.com/browse/HPCC-31103